tests/integration/lifecycle/test_craftctl.py

```diff
@@ -701,3 +701,37 @@ def test_craftctl_project_vars_write_once_from_state(new_dir, partitions, capfd)
     captured = capfd.readouterr()
     assert "variable 'myvar' can be set only once" in captured.err
     assert lf.project_info.get_project_var("myvar2") == "val2"
+
+
+def test_craftctl_chroot(new_dir, partitions, mocker):
```

Comment:
Patching out chroot etc. makes this look like a unit test to me. Could you make an integration test that runs a real command inside the chroot?